src/cdk-experimental/listbox/listbox.ts

```diff
@@ -6,19 +6,62 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive} from '@angular/core';
+import {
+  ContentChildren,
+  Directive,
+  EventEmitter, forwardRef, Inject,
+  Input, Output,
+  QueryList
+} from '@angular/core';
+import {BooleanInput, coerceBooleanProperty} from '@angular/cdk/coercion';
 
+/**
+ * Directive that applies interaction patterns to an element following the aria role of option.
+ * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
+ * value is built in.
+ */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
+    '(click)': 'toggle()',
+    '[attr.aria-selected]': '_selected || null',
+    '[attr.optionid]': '_optionid',
```

Comment:
It does need to be the native `id` property since `aria-activedescendant` relies on` id`. You can look at `MatCheckbox` for an example how we do do this, but by making `id` an `@Input` and assigning it an initial value, you ensure that it always has an ID while also letting the user set one if they need to for some reason (but people generally don't want to, that's why they're using a common component).

Reply:
I see I think `select` does that as well. I'll fix this.